File: Domain/Driver.cs
using System;

namespace Domain
{
    public class Driver
    {
        public Guid id { get; set; }
        public string name { get; set; }
        public string addres1 { get; set; }
        public string addres2 { get; set; }
        public string city { get; set; }
        public string postCode { get; set; }
        public int telphone { get; set; }
    }
}

File: school-run-client/src/app/models/driver.ts
export interface IDriver {
  id: string;
  name: string;
  addres1: string;
  addres2: string;
  city: string;
  postCode: string;
  telphone: string;
}

export interface IDriverFormValues extends Partial<IDriver> {
  telphone?: string;
}

export class DriverFormValues implements IDriverFormValues {
  id?: string = undefined;
  name: string = "";
  addres1: string = "";
  addres2: string = "";
  city: string = "";
  postCode: string = "";
  telphone?: string = undefined;

  constructor(init?: IDriverFormValues) {
    // if (init && init.telphone) {
    //   init.telphone = init.telphone;
    // }
    Object.assign(this, init);
  }
}
